aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1348 @@
+    ///
+    /// This can only be invoked by the multisig account itself, through the proposal flow.
+    /// Note that this function is not public so it can only be invoked directly instead of via a module or script. This
+    /// ensures that a multisig transaction cannot lead to another module obtaining the multisig signer and using it to
+    /// maliciously alter the number of signatures required.
+    entry fun update_signatures_required(
+        multisig_account: &signer, new_signatures_required: u64) acquires MultisigAccount {
+        let multisig_address = address_of(multisig_account);
+        assert_multisig_account_exists(multisig_address);
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(multisig_address);
+        // Short-circuit if the new number of signatures required is the same as before.
+        // This avoids emitting an event.
+        if (multisig_account_resource.signatures_required == new_signatures_required) {
+            return
+        };
+        let num_owners = vector::length(&multisig_account_resource.owners);
+        assert!(
+            new_signatures_required > 0 && new_signatures_required <= num_owners,
+            error::invalid_argument(EINVALID_SIGNATURES_REQUIRED),
+        );
+
+        let old_signatures_required = multisig_account_resource.signatures_required;
+        multisig_account_resource.signatures_required = new_signatures_required;
+        emit_event(
+            &mut multisig_account_resource.update_signature_required_events,
+            UpdateSignaturesRequiredEvent {
+                old_signatures_required,
+                new_signatures_required,
+            }
+        );
+    }
+
+    /// Create a multisig transaction, which will have one approval initially (from the creator).
+    ///
+    /// @param target_function The target function to call such as 0x123::module_to_call::function_to_call.
+    /// @param args Vector of BCS-encoded argument values to invoke the target function with.
+    public entry fun create_transaction(
+        owner: &signer,
+        multisig_account: address,
+        payload: vector<u8>,
+    ) acquires MultisigAccount {
+        assert!(vector::length(&payload) > 0, error::invalid_argument(EPAYLOAD_CANNOT_BE_EMPTY));
+
+        assert_multisig_account_exists(multisig_account);
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(multisig_account);
+        assert_is_owner(owner, multisig_account_resource);
+
+        let creator = address_of(owner);
+        let transaction = MultisigTransaction {
+            payload: option::some(payload),
+            payload_hash: option::none<vector<u8>>(),
+            approvals: simple_map::create<address, bool>(),
+            rejections: simple_map::create<address, bool>(),
+            creator,
+            metadata: simple_map::create<String, vector<u8>>(),
```

Comment:
Shall we allow application defined metadata values?

Reply:
My thought is this is meant primarily for display purposes. Whether these should be used for beyond display, including system attributes such as expiration time is still up for debate and would warrant more design to make sure it's safe.